src/FastConv/FastConv.cpp

```diff
@@ -10,23 +11,259 @@ CFastConv::~CFastConv( void )
+    {
+        float ac =  m_pfRealInput[i] * m_pfRealIr[i];
+        float bc = m_pfImagInput[i] * m_pfRealIr[i];
+        float ad = m_pfRealInput[i] * m_pfRealIr[i];
+        float bd = m_pfImagInput[i] * m_pfImagIr[i];
+        m_pfRealConv[i] = ac - bd;
+        m_pfImagConv[i] = ad + bc;
+    }
+    m_pCFftInstance->mergeRealImag(m_pfFreqConv, m_pfRealConv, m_pfImagConv);
+    // Convert back to time domain
+    m_pCFftInstance->doInvFft(m_pfConvOuput, m_pfFreqConv);
+}
+
+
+
+
+Error_t CFastConv::conv()
+{
+    //clear previous values in m_pfConvMatrix
+    for (int i=0; i<m_iBlockLength*(m_iBlockNum+1); i++)
+        m_pfConvMatrix[m_iCirIdx][i] = 0.f;
+    
+    for (int i=0; i<m_iBlockLength; i++)
+        m_pfOutputBlock[i] = 0.f;
+    
+    for (int i=0; i<m_iBlockLength*2; i++)
+        m_pfBlockConvOuput[i] = 0.f;
+    
+    
+    for (int i = 0; i<m_iBlockNum; i++)
+    {
+        //switch to freqConv() after implemented
+        switch (m_eCompMode) {
+            case kTimeDomain:
+                timeConv(i);
+                break;
+            case kFreqDomain:
+                freqConv(i);
+                break;
+            case kNumConvCompModes:
+                break;
+        }
+        
+        
+        for (int n = 0; n < m_iBlockLength*2 - 1; n++)
+            m_pfConvMatrix[m_iCirIdx][n+i*m_iBlockLength] += m_pfBlockConvOuput[n];
+    }
+    
+    for (int i = 0; i < m_iBlockNum+1; i++)
+    {
+        int iMatrixIdx = (m_iCirIdx+m_iBlockNum+1-i) % (m_iBlockNum+1);
+        for (int j = 0; j < m_iBlockLength; j++)
+            m_pfOutputBlock[j] += m_pfConvMatrix[iMatrixIdx][i*m_iBlockLength+j];
+    }
+    
     return Error_t::kNoError;
 }
+ 
 
-Error_t CFastConv::process (float* pfOutputBuffer, const float *pfInputBuffer, int iLengthOfBuffers )
+Error_t CFastConv::process (float* pfOutputBuffer, const float *pfInputBuffer, int iLengthOfBuffers)
 {
+    if (!m_bIsInitialized)
+    {
+        std::cout<< "not initialized";
+        return Error_t::kNotInitializedError;
+    }
+    
+    
+    m_iReset = 0;
+    
+    for (int i=0; i<iLengthOfBuffers; i++)
+    {
+        m_pfInputBlock[i-m_iReset] = pfInputBuffer[i];
+
+        if ((i+1) % m_iBlockLength == 0)
+        {
+            //output in pfBlockConvOuput
+            conv();
+            
+            for (int n=0; n<m_iBlockLength; n++)
+                pfOutputBuffer[n+m_iReset] = m_pfOutputBlock[n];
+            
+            m_iReset += m_iBlockLength;
+            
+            if (++m_iCirIdx >= m_iBlockNum+1)
+                m_iCirIdx = 0;
+        }
+    }
+    
+    
     return Error_t::kNoError;
 }
 
 Error_t CFastConv::flushBuffer(float* pfOutputBuffer)
 {
+    
+    if (!m_bIsInitialized)
+    {
+        std::cout<< "not initialized";
+        return Error_t::kNotInitializedError;
+    }
+
+    
+    for (int i=0; i<m_iBlockLength*2; i++)
+        m_pfInputBlock[i] = 0.f;
+    
+    for (int i=0; i<m_iLengthOfIr; i++)
```

Comment:
Flush buffer length should be different for time domain and frequency domain convolution